src/global-firestore-data/courses.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  coursesCollection,
+  availableRostersForCoursesCollection,
+  crseIdToCatalogNbrCollection,
+} from '../firebase-config';
+
+/**
+ * This function uses the subject and number of a course to retrieve the course information from 'courses' collection
+ * @param season The season that the course is under in the user's semester view
+ * @param year The year the course is under in the user's semester view
+ * @param subject The subject of the course
+ * @param number The number of the course
+ * @returns `Promise<CornellCourseRosterCourseFullDetail>`
+ */
+export const getCourseWithSeasonAndYear = async (
+  season: FirestoreSemesterSeason,
+  year: number,
+  subject: string,
+  number: string
+): Promise<CornellCourseRosterCourseFullDetail> => {
+  const roster = seasonAndYearToRosterIdentifier(season, year);
+  const course = await getDoc(doc(coursesCollection, `${roster}/${subject}/${number}`));
+  if (!course.exists()) {
+    const availableRostersForCourse = (
+      await getDoc(doc(availableRostersForCoursesCollection, `${subject} ${number}`))
+    ).data() as { rosters: string[] };
+    const lastRoster = availableRostersForCourse.rosters.length - 1;
+    const latestCourse = await getDoc(
+      doc(
+        coursesCollection,
+        `${availableRostersForCourse.rosters[lastRoster]}/${subject}/${number}`
+      )
+    );
+    return latestCourse.data()?.course as CornellCourseRosterCourseFullDetail;
+  }
+  return course.data()?.course as CornellCourseRosterCourseFullDetail;
+};
+
+/**
+ * This function uses the crseId and desired roster of a course to retrieve the course information from 'courses' collection
+ * @param roster The roster from which we want to fetch the course
+ * @param crseId  The course id of the course
+ * @returns `Promise<CornellCourseRosterCourseFullDetail>`
+ */
+export const getCourseWithCrseIdAndRoster = async (
+  roster: string,
+  crseId: number
+): Promise<CornellCourseRosterCourseFullDetail> => {
+  // use crseId to retrieve course subject and code
+  const courseSubjectAndNumber = await getDoc(doc(crseIdToCatalogNbrCollection, `${crseId}`));
+  const subject = courseSubjectAndNumber.data()?.split(' ')[0];
+  const number = courseSubjectAndNumber.data()?.split(' ')[1];
+
+  const course = await getDoc(doc(coursesCollection, `${roster}/${subject}/${number}`));
+  if (!course.exists()) {
+    const availableRostersForCourse = (
+      await getDoc(doc(availableRostersForCoursesCollection, `${subject} ${number}`))
+    ).data() as { rosters: string[] };
+    const lastRoster = availableRostersForCourse.rosters.length - 1;
+    const latestCourse = await getDoc(
+      doc(
+        coursesCollection,
+        `${availableRostersForCourse.rosters[lastRoster]}/${subject}/${number}`
+      )
+    );
+    return latestCourse.data()?.course as CornellCourseRosterCourseFullDetail;
+  }
+  return course.data()?.course as CornellCourseRosterCourseFullDetail;
+};
```

Comment:
Do you think it would be useful to look up a function by a roster like this, i.e. by a string like `FA20`? I feel like it might expose too much of the internal details of our Firestore schema and could also be error-prone

Reply:
I do agree with what you are saying, but don't we already expose this since the BottomBar says "SP23 Lecture Information"?

<img width="491" alt="Screen Shot 2023-03-25 at 4 54 37 PM" src="https://user-images.githubusercontent.com/46412997/227741281-f8322607-9a25-4470-b340-bba43b78ec67.png">

Like in this case. Let me know if what you are talking about is something different!

Reply:
Ah I see now, nevermind.

Reply:
For now, I will remove `getCourseWithCrseIdAndRoster()` and we can re-implement it when we need it! For Pablo's work this week, I don't think it will be necessary.